aspnetcore/security/authentication/jwt-authn.md

```diff
@@ -0,0 +1,155 @@
+---
+title: JWT authentication and 
+author: rick-anderson
+description: Learn how to set up manage JSON Web Tokens in development with dotnet user-jwts
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.date: 09/22/2018
+ms.custom: "mvc, seodec18"
+uid: security/authentication/jwt
+---
+
+# Manage JSON Web Tokens in development with dotnet user-jwts
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+The `dotnet user-jwts` command line tool can create and manage app specific local [JSON Web Tokens](https://jwt.io/introduction) (JWTs).
+
+## Name
+
+`dotnet user-jwts` - Cleans the output of a project.
+
+## Synopsis
+
+```dotnetcli
+dotnet user-jwts [<PROJECT>|<SOLUTION>] [command]
+dotnet user-jwts [command] -h|--help
+```
+
+## Description
+
+Create and manages app specific local JSON Web Tokens.
+
+## Arguments
+
+`PROJECT | SOLUTION`
+
+The MSBuild project or solution to apply a command on. If a project or solution file is not specified, MSBuild searches the current working directory for a file that has a file extension that ends in *proj* or *sln*, and uses that file.
+
+## Commands
+
+| Command  | Description |
+| ------------- | ------------- |
+| clear  |  Delete all issued JWTs for a project. |
+| create | Issue a new JSON Web Token.   |
+| delete | Delete a given JWT. |
+| key | Display or reset the signing key used to issue JWTs. |
+| list | Lists the JWTs issued for the project. |
+| print | Display the details of a given JWT. |
+
+### Create
+
+Usage: `dotnet user-jwts create [options]`
+
+| Option  | Description |
+| ------------- | ------------- |
+|  -p \| --project | The path of the project to operate on. Defaults to the project in the current directory. |
+| --scheme | The scheme name to use for the generated token. Defaults to 'Bearer'. |
+| -n \| --name | The name of the user to create the JWT for. Defaults to the current environment user. |
+| --audience | The audiences to create the JWT for. Defaults to the URLs configured in the project's launchSettings.json. |
```

Comment:
Is it really `launchSettings.json`?
In the example below they are given in `appsettings.Development.json`.

Reply:
> Is it really launchSettings.json?

Yep.

> In the example below they are given in appsettings.Development.json.

We parse launch settings for the ports that your app is served on and then add those to `appsettings.Development.json` under the `Authentication:Schemes:Bearer:Audience` category so they can be consumed by the options-from-config infrastructure at runtime.